packages/expo-splash-screen/ios/EXSplashScreen/EXSplashScreenService.m

```diff
@@ -0,0 +1,124 @@
+  }
+  
+  return [[_splashScreenControllers objectForKey:viewController] preventAutoHideWithCallback:successCallback
+                                                                             failureCallback:failureCallback];
+}
+
+- (void)hide:(UIViewController *)viewController successCallback:(void (^)(void))successCallback
+                                                failureCallback:(void (^)(NSString * _Nonnull))failureCallback
+{
+  if (![_splashScreenControllers objectForKey:viewController]) {
+    return failureCallback(@"No Native Splash Screen registered for given ViewController. First call 'SplashScreen.show' for given ViewController.");
+  }
+  return [[_splashScreenControllers objectForKey:viewController] hideWithCallback:successCallback
+                                                                  failureCallback:failureCallback];
+}
+
+- (void)onAppContentDidAppear:(UIViewController *)viewController
+{
+  if (![_splashScreenControllers objectForKey:viewController]) {
+    UMLogWarn(@"No Native Splash Screen registered for given ViewController. First call 'SplashScreen.show' for given ViewController.");
+  }
+  [[_splashScreenControllers objectForKey:viewController] onAppContentDidAppear];
+}
+
+- (void)onAppContentWillReload:(UIViewController *)viewController
+{
+  if (![_splashScreenControllers objectForKey:viewController]) {
+    UMLogWarn(@"No Native Splash Screen registered for given ViewController. First call 'SplashScreen.show' for given ViewController.");
+  }
+  [[_splashScreenControllers objectForKey:viewController] onAppContentWillReload];
+}
+
+# pragma mark - UIApplicationDelegate
+
+- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions
+{
+  // Read SplashScreen.plist
+  EXSplashScreenImageResizeMode resizeMode = EXSplashScreenImageResizeMode_CONTAIN;
+  NSString *plistPath = [[NSBundle mainBundle] pathForResource:@"SplashScreen" ofType:@"plist"];
+  if (plistPath) {
+    NSDictionary *splashScreenSettings = [NSDictionary dictionaryWithContentsOfFile:plistPath];
+    NSString *resizeModeString = splashScreenSettings[@"ResizeMode"];
+    resizeMode = [EXSplashScreenService resizeModeFromString:resizeModeString];
+  } else {
+    UMLogWarn(@"Cannot locate SplashScreen.plist!. Please install 'expo-splash-screen' properly.");
+  }
+  
+  UIViewController *rootViewController = [[UIApplication.sharedApplication keyWindow] rootViewController];
+  
+  [self show:rootViewController resizeMode:resizeMode];
+  return YES;
+}
+
+# pragma mark - static helpers
+
++ (EXSplashScreenImageResizeMode)resizeModeFromString:(NSString *)resizeMode
+{
+  resizeMode = [resizeMode lowercaseString];
+  if (resizeMode == nil || [resizeMode isEqualToString:@"contain"]) {
+    return EXSplashScreenImageResizeMode_CONTAIN;
+  } else if ([resizeMode isEqualToString:@"cover"]) {
```

Comment:
Since `contain` is the default one, maybe we could minify this to just one `if`?